docs/spec/spec.md

```diff
@@ -0,0 +1,310 @@
+
+1: Kubernetes type.
+
+#### Metadata
+
+##### Owner References
+
+- Owns EventTypes.
+
+#### Status
+
+| Field       | Type                      | Description                                                          | Limitations                                                                    |
+| ----------- | ------------------------- | -------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
+| provisioned | []ProvisionedObjectStatus | Status of creation or adoption of each EventType and errors therein. | It is expected that a provisioner list all produced EventTypes, if applicable. |
+
+#### Events
+
+- Source created
+- Source deleted
+- Event types installed
+
+### Life Cycle
+
+| Action | Reactions                                                                       | Limitations                                                                                              |
+| ------ | ------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------- |
+| Create | Creates and owns EventTypes produced, or adds Owner ref to existing EventTypes. | Verifies Json Schema provided by existing EventTypes; Not allowed to edit EventType if previously Owned; |
+| Update | Synchronizes EventTypes.                                                        |                                                                                                          |
+| Delete | Removes Owner ref from EventTypes.                                              |                                                                                                          |
+
+---
+
+## kind: Subscription
+
+### group: eventing.knative.dev/v1alpha1
+
+_Describes a linkage between a Subscribable and a Targetable and/or Sinkable._
+
+### Object Schema
+
+#### Spec
+
+| Field              | Type         | Description                                                                  | Limitations        |
+| ------------------ | ------------ | ---------------------------------------------------------------------------- | ------------------ |
+| from\*             | ObjectRef    | The originating Channel for the link.                                        | Must be a Channel. |
+| call<sup>1</sup>   | EndpointSpec | Optional processing on the event. The result of call will be sent to result. |                    |
+| result<sup>1</sup> | ObjectRef    | The continuation Channel for the link.                                       | Must be a Channel. |
+
+\*: Required
+
+1: At Least One(call, result)
+
+#### Metadata
+
+##### Owner References
+
+- If a resource controller created this Subscription: Owned by the originating
+  resource.
+
+##### Conditions
+
+- **Ready.**
+- **FromReady.**
+- **CallActive.** True if the call is sinking events without error.
```

Comment:
How does the subscription controller know this? This does seem useful, but I'd like to make sure we know how we're going to get that data out of the data plane, through the channel, and up to the subscription.

Reply:
At a technical level, the event dispatcher for the channel is the only place with immediate knowledge that the dispatched failed. It's possible the service mesh could also be aware, but that's a bigger stretch. Either way, it's a bit roundabout to update the subscription's status.

At a semantic level, the call doesn't sink events, the result sinkable.

Moreover, the definitions is a vague. Should the value be true or false if the channel has no events to dispatch? How many failure over what period of time should cause the condition to flip?

Reply:
Will remove, need to fix the larger issue https://github.com/knative/eventing/issues/536